uportal-war/src/main/java/org/apereo/portal/security/provider/AnyUnblockedGrantPermissionPolicy.java

```diff
@@ -155,7 +155,7 @@ public boolean doesPrincipalHavePermission(
         boolean rslt;
         try {
             // Track groups we've already explored to avoid infinite loop
-            final Set<IGroupMember> seenGroups = new HashSet<IGroupMember>(100);
+            final Set<IGroupMember> seenGroups = new HashSet<>(100);
```

Comment:
> The core Java collection classes scale automatically, but specifying an initialCapacity is a way to provide a "hint" about what size you think the thing will probably need to support. If you do a good job estimating -- the theory goes -- the code will perform better.

`HashSet` grows in amortized constant time `O(1)`, which is pretty good.
Unless the magic number `100` has some meaning here (which should be documented), the initial allocation is probably slower than using the defaults letting `HashSet` grow on its own.

> I didn't change the initialCapacity, FWIW; I didn't apply fresh consideration as to whether this number is a good one or whether we should specify one at all.

Fair enough